### ISSUE TITLE
Ability to link cue sheets and CHD for Sega CD

### ISSUE THREAD

**Issue by televandalist** [RetroAchievements/RALibretro#115]
Sega CD games work fine, but hashes for the cue sheets or CHDs currently need to be added remotely. Using the Gens core and trying to start a game, RALib does not look for cue or CHD. Also, it associates bin files with Genesis/Mega Drive, likely because some Gen/MD ROMs are bin files. 

Caricatur, Shmelyoff, and myself have all tested Popful Mail and can confirm that it works.

**Comment by televandalist** [comment]
oh and apparently 32x works with PicoDrive 🤷‍♂

**Comment by Jamiras** [comment]
RAlibRetro doesn't know how to hash Sega CD games. Right now it's hashing the cue file, or the chd file (in its entirety). We probably want something akin to the PSX hashing where we hash the executable - more investigation is required in that regard.

Regarding 32x - the files are being fully hashed, and are roughly the same size as regular Genesis games, so they could be supported without any additional effort. I did notice rather glitchy graphics loading Pitfall The Mayan Adventure, so it's unclear how the core itself performs. It's probably safe to include 32x support in a future release.

**Comment by Jamiras** [comment]
Determined that Sega CD hashing will have to be very simplistic. The volume header and ROM header are always present in the same location, but they're followed by two arbitrary chunks of code. The first ensures the hardware region matches the CD region. The second loads the primary executable of the game. Without processing the arbitrary code, it's impossible to determine which file is the primary executable, and most of the games I examined appeared to have multiple executables for different portions of the game. As such, it's unreasonable to identify and hash the executable code. As a result, only the volume and ROM headers are hashed (the first 512 bytes of sector 0).

cue and iso files are supported by the Genesis Plus GX core, and can be handled by RALibRetro. 

RALibRetro would handle the m3u file, but Genesis Plus GX does not. Conversely, Genesis Plus GX handles the chd, but RALibRetro does not.

Implemented in #117.
